### PR TITLE
sage: 9.3 -> 9.4

### PR DESCRIPTION
**Pull request by collares**
###### Motivation for this change

Tracking upstream changes so we are ready when 9.4 gets released. As of July 31st, tests for 9.4.rc0 pass.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/memory_allocator/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ lib
+, fetchPypi
+, buildPythonPackage
+, cython
+}:
+
+buildPythonPackage rec {
+  pname = "memory_allocator";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/memory_allocator/default.nix:8]:
```suggestion
  pname = "memory-allocator";
```

File: pkgs/development/python-modules/memory_allocator/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-UUcR71e3eAQIQpmWM+AVQxVtgHvrNjaIlHo5pURUln0=";
+  };
+
+  propagatedBuildInputs = [
+    cython
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/memory_allocator/default.nix:19]: checkInputs and pythonImportsCheck missing.

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -4409,6 +4409,8 @@ in {
 
   memcached = callPackage ../development/python-modules/memcached { };
 
+  memory_allocator = callPackage ../development/python-modules/memory_allocator { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:4412]:
```suggestion
  memory-allocator = callPackage ../development/python-modules/memor-allocator { };
```

File: pkgs/development/python-modules/memory_allocator/default.nix (hunk 2 above)
Comment (collares) [reply pkgs/development/python-modules/memory_allocator/default.nix:19]: Thanks for the heads-up! I looked at this; this package has tests, and they are executed successfully when running `python setup.py test` (so the default `setuptoolsCheckPhase` runs the tests too). The tests are written in a way that  `pytest` (via `pytestCheckHook`) does not find anything to run. I'm open to suggestions, but I think leaving this implicit runs the most tests.

**Comment by collares** [comment]
Result of `nixpkgs-review pr 130396` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>maxima</li>
    <li>maxima-ecl</li>
    <li>pynac</li>
    <li>python38Packages.memory-allocator</li>
    <li>python39Packages.memory-allocator</li>
    <li>sage (sagemath)</li>
    <li>sageWithDoc</li>
    <li>wxmaxima</li>
  </ul>
</details>

File: pkgs/development/python-modules/memory-allocator/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  src = fetchPypi {
+    inherit version;
+    pname = "memory_allocator";
+    sha256 = "sha256-UUcR71e3eAQIQpmWM+AVQxVtgHvrNjaIlHo5pURUln0=";
+  };
+
+  propagatedBuildInputs = [
+    cython
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/memory-allocator/default.nix:19]: Missing checkInputs and pythonImportsCheck.

Comment (collares) [reply pkgs/development/python-modules/memory-allocator/default.nix:19]: See above :)

**Comment by collares** [comment]
Result of `nixpkgs-review pr 130396` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>maxima</li>
    <li>maxima-ecl</li>
    <li>pynac</li>
    <li>python38Packages.memory-allocator</li>
    <li>python39Packages.memory-allocator</li>
    <li>sage (sagemath)</li>
    <li>sageWithDoc</li>
    <li>wxmaxima</li>
  </ul>
</details>

**Comment by collares** [comment]
ofBorg seems happy, both on x86_64 and on aarch64.

File: pkgs/development/python-modules/memory-allocator/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/memory-allocator/default.nix:19]: pythonImportsCheck should be added anyway.

Comment (collares) [reply pkgs/development/python-modules/memory-allocator/default.nix]: OK, added. But, to be clear, the package does have a test suite comprised of 35 tests, and the default `setuptoolsCheckPhase` causes those tests to be executed.

**Comment by collares** [comment]
Thanks for merging!

**Comment by timokau** [comment]
Thank you once again for keeping sage running and up-to-date :)

**Pull request merged**